src/main/scala/io/moia/protos/teleproto/ReaderImpl.scala

```diff
@@ -61,10 +53,10 @@ class ReaderImpl(val c: blackbox.Context) extends FormatImpl {
     *
     * The result is `f` applied to the reader expression with the (possible) compatibility issues of reader generation (if happened).
     */
-  private def withImplicitReader(protobufType: Type, modelType: Type)(compileInner: Tree => Tree): Compiled = {
+  private def withImplicitReader[P, M](protobufType: Type[P], modelType: Type[M])(compileInner: Tree => Tree)(using Quotes): Compiled = {
```

Comment:
I think at this point we should be working with `TypeRepr` instead of `Type`, right?